File: Rnwood.Smtp4dev/ApiClient/SessionSummary.ts
 

export default class SessionSummary {

    constructor(clientAddress: string, clientName: string, numberOfMessages: number, id: string, endDate: Date, ) {
         
        this.clientAddress = clientAddress; 
        this.clientName = clientName; 
        this.numberOfMessages = numberOfMessages; 
        this.id = id; 
        this.endDate = endDate;
    }

     
    clientAddress: string; 
    clientName: string; 
    numberOfMessages: number; 
    id: string; 
    endDate: Date;
}

File: Rnwood.Smtp4dev.Shared/ApiModel/SessionSummary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Rnwood.Smtp4dev.ApiModel
{
    public class SessionSummary : ICacheByKey
    {

        public string ClientAddress { get; set; }
        public string ClientName { get; set; }
        public int NumberOfMessages { get; set; }
        public Guid Id { get; set; }
        public DateTime? EndDate { get; set; }
        public DateTime StartDate { get; set; }
        public bool TerminatedWithError { get; set; }

        public int Size { get; set; }

        string ICacheByKey.CacheKey => Id.ToString();
    }
}
